lsp/core/aws-lsp-core/src/content/uriResolverBuilder.ts

```diff
@@ -1,22 +1,22 @@
 import { URI } from 'vscode-uri'
 import { ContentRequestMiddleware, ContentRequestMiddlewareDelegate } from './contentRequestMiddleware'
-import { SchemaProvider } from './schemaProvider'
+import { UriResolver } from './uriResolver'
 
 /**
- * Creates a middleware pipeline for resolving requests for a schema.
+ * Creates a middleware pipeline for resolving requests for content from a uri.
  * Middleware handlers will be called in the order they are added, until a
  * handler returns a response. At this point, the handlers in the call stack
  * have a chance to handle/process the response.
  */
-export class SchemaProviderBuilder {
+export class UriResolverBuilder {
```

Comment:
Why does a "resolver" need a "builder", can the indirection be eliminated?

Reply:
The builder is a way of creating a middleware pipeline that chains uri resolvers together into a composite UriResolver. You can use UriResolver without the builder though. Open to suggestions for alternative names for UriResolverBuilder.

Reply:
can the resolver be composable without being wrapped in a builder?

Reply:
Not very nicely. The code would look more complicated [in places where you would define the composition](https://github.com/aws/aws-toolkit-common/pull/510/files#diff-3bc3c866c395540756bac81fec325f2fa410073ce82467181fe68a65a723ad8fR24), and at the place where we [hand off the resolver into third party code](https://github.com/aws/aws-toolkit-common/pull/510/files#diff-5b6b83b0b1531d2a705ec63b9fd21cadcddad7c6b60375f62458a46073ccf6e2R20)